Cesium3DTilesContent/src/I3dmToGltfConverter.cpp

```diff
@@ -0,0 +1,777 @@
+    return instances;
+  }
+  auto getInstanceAccessor = [&](const char* name) -> const Accessor* {
+    if (auto accessorItr = gpuExt.attributes.find(name);
+        accessorItr != gpuExt.attributes.end()) {
+      return Model::getSafe(&model.accessors, accessorItr->second);
+    }
+    return nullptr;
+  };
+  auto errorOut = [&](const std::string& errorMsg) {
+    errors.emplaceError(errorMsg);
+    return instances;
+  };
+
+  const Accessor* translations = getInstanceAccessor("TRANSLATION");
+  const Accessor* rotations = getInstanceAccessor("ROTATION");
+  const Accessor* scales = getInstanceAccessor("SCALE");
+  int64_t count = 0;
+  if (translations) {
+    count = translations->count;
+  }
+  if (rotations) {
+    if (count == 0) {
+      count = rotations->count;
+    } else if (count != rotations->count) {
+      return errorOut(fmt::format(
+          "instance rotation count {} not consistent with {}",
+          rotations->count,
+          count));
+    }
+  }
+  if (scales) {
+    if (count == 0) {
+      count = scales->count;
+    } else if (count != scales->count) {
+      return errorOut(fmt::format(
+          "instance scale count {} not consistent with {}",
+          scales->count,
+          count));
+    }
+  }
+  if (count == 0) {
+    return errorOut("No valid instance data");
+  }
+  instances.resize(static_cast<size_t>(count), glm::dmat4(1.0));
+  if (translations) {
+    AccessorView<CesiumGltf::AccessorTypes::VEC3<float>> translationAccessor(
+        model,
+        *translations);
+    if (translationAccessor.status() == AccessorViewStatus::Valid) {
+      for (unsigned i = 0; i < count; ++i) {
+        auto transVec = toGlm<glm::dvec3>(translationAccessor[i]);
+        instances[i] = glm::translate(instances[i], transVec);
+      }
+    } else {
+      return errorOut("invalid accessor for instance translations");
+    }
+  }
+  if (rotations) {
+    auto quatAccessorView = getQuaternionAccessorView(model, rotations);
+    std::visit(
+        [&](auto&& arg) {
+          for (unsigned i = 0; i < count; ++i) {
+            auto quat = toGlmQuat<glm::dquat>(arg[i]);
+            instances[i] = instances[i] * glm::toMat4(quat);
+          }
+        },
+        quatAccessorView);
+  }
+  if (scales) {
+    AccessorView<CesiumGltf::AccessorTypes::VEC3<float>> scaleAccessor(
+        model,
+        *scales);
+    if (scaleAccessor.status() == AccessorViewStatus::Valid) {
+      for (unsigned i = 0; i < count; ++i) {
+        auto scaleFactors = toGlm<glm::dvec3>(scaleAccessor[i]);
+        instances[i] = glm::scale(instances[i], scaleFactors);
+      }
+    } else {
+      return errorOut("invalid accessor for instance translations");
+    }
+  }
+  return instances;
+}
+
+const size_t rotOffset = sizeof(glm::vec3);
+const size_t scaleOffset = rotOffset + sizeof(glm::quat);
+const size_t totalStride =
+    sizeof(glm::vec3) + sizeof(glm::quat) + sizeof(glm::vec3);
+
+void copyToBuffer(
```

Comment:
Just to be more explicit

```suggestion
void copyInstanceToBuffer(
```